addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -2,17 +2,62 @@ odoo.define('website_mass_mailing.editor', function (require) {
 'use strict';
 
 var core = require('web.core');
+const Dialog = require('web.Dialog');
 var rpc = require('web.rpc');
 var options = require('web_editor.snippets.options');
 var wUtils = require('website.utils');
 
 const qweb = core.qweb;
 var _t = core._t;
 
+const mailingListSubscribe = {
 
-options.registry.mailing_list_subscribe = options.Class.extend({
-    popup_template_id: "editor_new_mailing_list_subscribe_button",
-    popup_title: _t("Add a Newsletter Subscribe Button"),
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Shows a warning popup, if mailing list is not exists in the database.
+     *
+     * @private
+     */
+     _showWarning() {
+        const text = _t("No mailing list exists in the database, Do you want to create a new mailing list!");
+        Dialog.confirm(this, text, {
+            title: _t("Warning!"),
+            confirm_callback: () => {
+                window.location.href = '/web#action=mass_mailing.action_view_mass_mailing_lists';
```

Comment:
This would quit the editor without saving anything.. needed to request save on editor event